src/Infrastructure/Services/FraudProtectionService.cs

```diff
@@ -100,84 +100,88 @@ private async Task<T> Read<T>(HttpResponseMessage response) where T : new()
             var content = await response.Content.ReadAsStringAsync();
             response.Dispose();
 
-            return JsonSerializer.Deserialize<T>(content, _responseDeserializationOptions);
+            return new SampleResponse<T>
+            {
+                Data = JsonSerializer.Deserialize<T>(content, _responseDeserializationOptions),
+                RawData = JsonSerializer.Deserialize<object>(content, _responseDeserializationOptions),
+            };
```

Comment:
here is where we Deserialize twice... once to the known type and once to a generic object... and we store both.